vitrina/datasets/migrations/0012_auto_20250804_0927.py

```diff
@@ -0,0 +1,52 @@
+# Generated by Django 3.2.25 on 2025-08-04 06:27
+
+from django.db import migrations
+
+def map_type_to_subclass(type_name):
+    mapping = {
+        'series': 'series',
+        'is': 'information_system',
+        'catalog': 'catalog',
+        'service': 'service',
+    }
+    return mapping.get(type_name)
+
+def set_subclass_for_datasets(apps, schema_editor):
+    Dataset = apps.get_model('vitrina_datasets', 'Dataset')
+    Type = apps.get_model('vitrina_datasets', 'Type')
+    DCATResourceSubclass = apps.get_model('vitrina_datasets', 'DCATResourceSubclass')
+
+    dataset_subclass = DCATResourceSubclass.objects.get(
+        name='dataset'
+    )
+
+    for dataset in Dataset.objects.all():
+        if dataset.subclass_id:
+            continue
+
+        types = dataset.type.all()
+        if types.exists():
+            first_type = types.first()
```

Comment:
```suggestion
            first_type = types[0]
```

`types.first()` will do additional query to DB. `types[0]` will not, if types are prefetched